packages/combobox/src/Combobox/Combobox.tsx

```diff
@@ -749,10 +750,15 @@ const Option = <T extends TgphElement>({
   const [isFocused, setIsFocused] = React.useState(false);
   const contextValue = context.value;
 
+  const textContent = React.useMemo(
+    () => coerceChildrenToText(children),
+    [children],
+  );
+
   const isVisible =
     !context.searchQuery ||
     doesOptionMatchSearchQuery({
-      label: label?.toString() || children?.toString(),
+      label: label?.toString() || textContent,
```

Comment:
if this is an OR condition based on the label existing, should we do this check before running the useMemo above? Otherwise the `coerceChildrenToText` function would run each time, even if we don't have to